workflow/controller/operator.go

```diff
@@ -529,7 +530,22 @@ func (woc *wfOperationCtx) persistUpdates() {
 	// conflicts arose when attempting to update, causing our conflict resolution code to be invoked. The conflict
 	// resolution code was not perfect and workflow information was not correctly persisted, causing undefined behavior.
 	// This line was reintroduced in v2.9.5, and should remain as long as we use our current informer pattern.
-	time.Sleep(1 * time.Second)
+	if os.Getenv("INFORMER_WRITE_BACK") == "true" {
```

Comment:
I think that having this off by default is a good idea. It has cons - we have an untested path of code, which mean it'll rot at some point.

If we can create a test that we can be confident if that we have fixed this, I'd suggest we flip this guard.